platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/AssignImmutableWorkspaceStep.java

```diff
@@ -146,10 +144,18 @@ private Optional<WorkspaceResult> loadImmutableWorkspaceIfConsistent(UnitOfWork
         ImmutableListMultimap<String, HashCode> outputHashes = calculateOutputHashes(outputSnapshots);
         ImmutableWorkspaceMetadata metadata = workspaceMetadataStore.loadWorkspaceMetadata(immutableLocation);
         if (!metadata.getOutputPropertyHashes().equals(outputHashes)) {
-            return workspace.withTemporaryWorkspace(temporaryWorkspace -> {
-                moveInconsistentImmutableWorkspaceToTemporaryLocation(immutableLocation, temporaryWorkspace, outputSnapshots);
-                return Optional.empty();
-            });
+            fileSystemAccess.invalidate(ImmutableList.of(immutableLocation.getAbsolutePath()));
+            String actualOutputHashes = outputSnapshots.entrySet().stream()
+                .map(entry -> entry.getKey() + ":\n" + entry.getValue().roots()
+                    .map(AssignImmutableWorkspaceStep::describeSnapshot)
+                    .collect(Collectors.joining("\n")))
+                .collect(Collectors.joining("\n"));
+            throw new IllegalStateException(String.format(
```

Comment:
❓ What is the solution for users if this happen? Should we give some hint how to resolve this issue?

Reply:
I decided to merge as it is. If this will be an issue users should open an issue to Gradle